nda-transform/count_assoc_files.py

```diff
@@ -1,81 +1,122 @@
-    if id=='0' or collection=='0':
-        _data_file_path = source_path
-        df = pd.read_csv(file, header=1)
-    else:
-        df = pd.read_csv(file)
-        _data_file_path = nda_submissions_path / df.loc[0]["SUBMISSION_FOLDER_NAME"]
-
+from typing import Dict, List, Literal, Optional, Set
 
-    # check a file only if at least one file_columns exist
-    exist=0
-    for c in file_columns:
-        if c in df.columns:
-            exist=1
-            break
-    if not exist:
-        continue
+import pandas as pd
 
 
-    for idx, row in df.iterrows():
-        
+def count_files(
+    project_path: Optional[Path],
+    source_path: Path,
+    files: List[Path],
+    context: Literal["aws", "eris"],
+):
+    """
+    Count unique and duplicate files in the given list of files
+    """
+    files_count: int = 0
+    unique_files: Set = set()
+    duplicate_files_count: Dict[str, int] = {}
+
+    file_columns = (
+        "data_file1 data_file2 image_file bvalfile bvecfile transcript_file".split()
+    )
+
+    for file in files:
+        print(f"Checking {file}")
+
+        if context == "eris":
+            _data_file_path = source_path
+            df = pd.read_csv(file, header=1)
+        else:
+            nda_submissions_path = project_path / "nda-submissions"
+            df = pd.read_csv(file)
+            _data_file_path = nda_submissions_path / df.loc[0]["SUBMISSION_FOLDER_NAME"]
+
+        # check a file only if at least one file_columns exist
+        check_file = False
         for c in file_columns:
-        
             if c in df.columns:
-                if not pd.isna(row[c]):
-                    data_file = row[c]
+                check_file = True
+                break
 
-                    data_file_path: Path = _data_file_path / data_file
+        if not check_file:
+            continue
 
-                    if not data_file_path.exists():
-                        print(f"File does not exist: {data_file_path}")
-                    else:
-                        count+=1
+        for _, row in df.iterrows():
+            for c in file_columns:
+                if c in df.columns:
+                    if not pd.isna(row[c]):
+                        data_file = row[c]
+                        data_file_path: Path = _data_file_path / data_file
 
-                        if data_file_path not in unique_file:
-                            unique_file[data_file_path]=1
+                        if not data_file_path.exists():
+                            print(f"File does not exist: {data_file_path}")
                         else:
-                            # count duplicates
-                            manifest=file.name
-                            if manifest in duplicate:
-                                duplicate[manifest]+=1
+                            files_count += 1
+                            if data_file_path not in unique_files:
+                                unique_files.add(data_file_path)
```

Comment:
Dheshan, I believe it is the fastest in Python to build a set by using `dict[a]=''` method. How well do you know about `Set()` method?